csharp/private/rules/resgen.bzl

```diff
@@ -0,0 +1,31 @@
+
+def _csharp_resx_impl(ctx):
+    print("todo")
+
+csharp_resx = rule(
+    implementation = _csharp_resx_impl,
+    attrs = {
+        "src": attr.label(
+            doc = "The resx file."
+            mandatory = True, 
+            allow_single_file = True
+        ),
+        "identifier": attr.string(

```

Comment:
I've been looking into an `out` parameter to specify the name of the compiled resource.  With `EmbeddedResource`'s you can specify the property `LogicalName` that is used by [GenerateResource](https://docs.microsoft.com/en-us/visualstudio/msbuild/generateresource-task?view=vs-2019) to set the output name.

> The GenerateResource task uses the <LogicalName> metadata of an <EmbeddedResource> item to name the resource that is embedded in an assembly.
>
> Assuming that the assembly is named myAssembly, the following code generates an embedded resource named someQualifier.someResource.resources:

Based on my testing so far I haven't been successful with getting the compiled resource named based on the `LogicalName` property. I'll try testing this from visual studio, to get a better idea of how this is happening in that environment.